pycbc/psd/variation.py

```diff
@@ -160,3 +165,214 @@ def find_trigger_value(psd_var, idx, start, sample_rate):
     ind -= 1
     vals = psd_var[ind]
     return vals
+
+
+def mean_square(data, delta_t, short_stride, stride):
+    """ Calculate mean square of given time series once per stride
+
+    First of all this function calculate the mean square of given time
+    series once per short_stride. This is used to remove outliers due
+    to short glitches. Then, every seconds it computes the mean square
+    of the smoothed time series, within the stride.
+
+    Parameters
+    ----------
+    data : TimeSeries
+    delta_t : float
+        Duration of the time series
+    short_stride : float
+        Stride duration for outlier removal
+    stride ; float
+        Stride duration
+
+    Returns
+    -------
+    m_s: List
+        Mean square of given time series
+    """
+
+    srate = int(data.sample_rate)
+
+    # Calculate mean square of data once per short stride and remove
+    # the outliers
+    data = numpy.array(data)
+    short_ms = numpy.mean(data.reshape(-1, int(srate * short_stride)) ** 2,
+                          axis=1)
+    ave = 0.5 * (short_ms[2:] + short_ms[:-2])
```

Comment:
i'm not entirely sure why this quantity is calculated - can you describe a bit more what it should be doing?

Reply:
Every element of "ave" is the average of the closest elements of the corresponding index of "short_ms". Consider for example the element of the array "short_ms" with index n. The element in the n index of "ave" will be the average between the (n-1) and (n+1) elements of short_ms. This is then use to remove the outliers (or smoothing) short_ms

Reply:
ah, ok - then it is testing whether any element is an 'outlier' with respect to its two neighbours. perhaps then this should be commented together with the following line which defines an outlier as anything bigger than 2* the average of the 'neighbours'

Reply:
Is it fine if I describe it at the beginning of the function?

Reply:
yes, commenting in one place may be easier to understand than many small comments on different lines